core/src/main/java/org/radargun/stages/test/RunningTest.java

```diff
@@ -0,0 +1,290 @@
+   private int minWaitingThreads;
+   private int maxThreads;
+   private boolean logTransactionExceptions;
+   private long minThreadCreationDelay;
+
+   public static String nameFor(String testName) {
+      return RunningTest.class.getName() + "." + testName;
+   }
+
+   public void setStatisticsPrototype(Statistics statisticsPrototype) {
+      this.statisticsPrototype = statisticsPrototype;
+   }
+
+   public boolean isSteadyState() {
+      return steadyState;
+   }
+
+   public void setSteadyState(boolean steadyState) {
+      this.steadyState = steadyState;
+   }
+
+   public boolean isFinished() {
+      return finished;
+   }
+
+   public boolean isTerminated() {
+      return terminated;
+   }
+
+   public void setTerminated() {
+      finished = true;
+      terminated = true;
+      steadyState = false;
+   }
+
+   public boolean isReachedMax() {
+      return reachedMax;
+   }
+
+   public Statistics createStatistics() {
+      if (statisticsPrototype == null) {
+         return null;
+      } else {
+         statisticsCounter.incrementAndGet();
+         return statisticsPrototype.newInstance();
+      }
+   }
+
+   public void recordStatistics(Statistics stats) {
+      statisticsQueue.add(stats);
+   }
+
+   public List<Statistics> getStatistics() {
+      interruptStressors();
+      ArrayList<Statistics> statistics = new ArrayList<>();
+      while (!terminated && statistics.size() < statisticsCounter.get()) {
+         Statistics stats = statisticsQueue.poll();
+         if (stats != null) {
+            statistics.add(stats);
+         } else {
+            try {
+               Thread.sleep(1000);
+            } catch (InterruptedException e) {
+               throw new IllegalStateException(e);
+            }
+         }
+      }
+      statisticsCounter.set(0);
+      return statistics;
+   }
+
+   protected void interruptStressors() {
+      synchronized (this) {
+         for (Stressor stressor : stressors) {
+            stressor.interrupt();
+         }
+      }
+   }
+
+   public void stopStressors() {
+      finished = true;
+      steadyState = false;
+      interruptStressors();
+      List<Stressor> stressors;
+      synchronized (this) {
+         stressors = new ArrayList<>(this.stressors);
+      }
+      for (Stressor stressor : stressors) {
+         try {
+            stressor.join();
+         } catch (InterruptedException e) {
+            throw new IllegalStateException(e);
+         }
+      }
+      synchronized (this) {
+         for (Runnable listener : stopListeners) {
+            listener.run();
+         }
+      }
+   }
+
+   public void addStressor(boolean failSilently) {
+      int threadIndex = getNextThreadId(maxThreads);
+      if (threadIndex < 0) {
+         if (failSilently) {
+            return;
+         }
+         reachedMax = true;
+         log.warnf("Attempt to create more than %d threads!", maxThreads);
+         Utils.threadDump();
+         return;
```

Comment:
Since this is called from another stressor, it does not make much sense to throw an exception and just catch it elsewhere, logging an error. The test will be later declared failed anyway.